src/main/java/org/spongepowered/asm/mixin/transformer/MixinConfig.java

```diff
@@ -504,6 +508,32 @@ public boolean hasMixinsFor(String targetClass) {
         }
         return mixins;
     }
+
+    /**
+     * Updates a mixin with new bytecode
+     *
+     * @param mixinClass Name of the mixin class
+     * @param bytes New bytecode
+     * @return List of classes that need to be updated
+     */
+    public List<String> reloadMixin(String mixinClass, byte[] bytes) {
+        for (Iterator<MixinInfo> iter = this.mixins.iterator(); iter.hasNext();) {
+            MixinInfo mixin = iter.next();
+            if (mixin.getClassName().equals(mixinClass)) {
+                mixin.reloadMixin(bytes);
+                try {
+                    mixin.validate();
+                } catch (Exception e) {
+                    this.logger.error(e.getMessage(), e);
```

Comment:
This was the catch I was talking about, potentially have this throw an error which can be caught and signalled to the instrumentation API.